ChessVariantsAPI/Controllers/UsersController.cs

```diff
@@ -0,0 +1,32 @@
+﻿using DataAccess.MongoDB;
+using DataAccess.MongoDB.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace ChessVariantsAPI.Controllers;
+[Route("api/[controller]")]
+[ApiController]
+public class UsersController : ControllerBase
+{
+
+    private readonly DatabaseService _db;
+
+    public UsersController(DatabaseService databaseService)
+    {
+        _db = databaseService;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<User>>> Get()
+    {
+        var users = await _db.Users.GetAsync();
+        return Ok(users);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<User>> Post(User user)
+    {
+        user = new User { Email = user.Email, FirstName = user.FirstName, LastName = user.LastName, UserName = user.UserName };
```

Comment:
Yeah it looks weird haha but it actually has a function. `User` has a MongoDB id field which is autogenerated on creation. However, the User which is passed as a parameter doesn't have this, and as such I recreate the object to create it! This will change soon though when I swap out the User parameter for a user-DTO.

Reply:
Ah okay, now I understand. Thanks!